### PR TITLE
♻️ REFACTOR: Reduce JS/CSS size and improve accessibility

### LINKED ISSUE #65: Unpleasant line under the tab names
Dear all,

I am using sphinx-tabs inside a [Jupyter Book](https://jupyterbook.org/) and I see this:
Screenshot 2020-05-11 at 20 06 17
where you can see a line dividing the tab and the content below it. Such line is not present in the examples shown [here](https://github.com/djungelorm/sphinx-tabs).

Any clues?

I could try to provide you with additional information.

Thank you.
-Filippo

### LINKED ISSUE #66: Add tabindex attribute to improve accessibility
If the [tabindex attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex) of `tabindex="0"` was added to each `a.item` anchor, the javascript could be modified slightly to allow keyboard control of the tabs.

I'll look at opening a PR for this, but wanted to create the issue first.

### LINKED ISSUE #80: Remember last `group-tab` selection between pages
**Is your feature request related to a problem? Please describe.**

I'd like the `group-tab` selection to persist between different pages. For example, consider a book containing code snippets in multiple languages. Users might like to select a particular language to reference throughout the book, rather than selecting that language's tab on each page.

**Describe the solution you'd like**

I'd considered storing the last `group-tab` selection in the session, so that it persists between pages (i.e. across a `jupyter-book`). Would be great to know if there's a more appropriate way to do this.

### LINKED ISSUE #87: RFE: Reduce project's JS/CSS payload size
**Is your feature request related to a problem? Please describe.**

Currently, this extension results in 50+ kB of assets across 7+ network requests to be added to each page for a project's documentation.

**Describe the solution you'd like**

Significantly lowered payload size (drop the dependency on semantic-ui?) and consolidate CSS/JS files to reduce the number of network requests needed.

**Describe alternatives you've considered**

**Additional context**

This would also make it easier for "downstream" themes to customize how the tabs look, if the CSS for this project is self-contained in a single file. :)

### LINKED ISSUE #90: Compatibility with pydata_sphinx_theme
* I would like to use `sphinx-tabs` in a project with `pydata_sphinx_theme` (https://github.com/pandas-dev/pydata-sphinx-theme). Unfortunately, they don't seem to be compatible.
* Using the example code together with the option `html_theme = 'pydata_sphinx_theme'` in the conf.py results in the following behavior.
image
* Do you have any idea how to achieve compatibility? Any help would be very much appreciated.

Versions used: python 3.8.2, pydata_sphinx_theme 0.3.2, sphinx-tabs 1.3.0, sphinx 3.2.1

### LINKED ISSUE #92: Improve accessibility of tabs
**Describe the bug**

Tabs can't be selected and switched using the keyboard alone.
This would help to comply with [W3C's web accessibility initiative](https://www.w3.org/WAI/WCAG21/quickref/#keyboard).

**To Reproduce**

Use the tab key to move through focusable objects on a page containing sphinx-tabs. Tab sets can't be selected or changed to show content on other tabs.

**Expected behavior**

Users should be able to highlight and switch tabs using the keyboard.
This behaviour is exhibited by [`sphinx-panels` tabbed content](https://sphinx-panels.readthedocs.io/en/latest/#tabbed-content), where the tab key can be used to select the tab set and arrow keys can be used to switch tab.

### LINKED ISSUE #93: Images should have a bottom margin of 24px
Or an alternative solution, see the image below for the awkward formatting of text in group-tabs alongside images.
image

### PR DESCRIPTION
**Pull request by foster999**
Removes:
* semantic-ui assets
* unused 'sphinx_tabs_nowarn' sphinx option

Changes:
* tab structure to use [tab roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Tab_Role)
* tab label colour, to increase contrast with background
* assets are now copied across by `sphinx` rather than explicitly in the extension

Adds:
* `tabindex` atributes to allow users to focus and switch tabs using a keyboard (left and right arrows, and enter key to change)
* ARIA labels for tabs and panels
* ability to close tabs panel by clicking open tab
* persistence of last selected group or code tab during the session (when session storage is permitted)
* margin below images inside tab content

Closes #92, closes #66, closes #87, closes #90 and closes #80, closes #65, closes #93.
It also includes an equivalent of #73

### CODE REVIEW

**Comment by foster999** [comment]
Tablist bottom border can appear half-covered. Try only setting tab background colour when selected

**Comment by foster999** [comment]
Quite a large change to the underlying HTML here, so wouldn't be surprised if it introduces bugs for some use cases... Though I've trialed it with the demos in the docs and it seems to work well.

Also worth noting that:
1. Tabs and tab panels will now be focused on clicking, giving them a black outline until another element is focused. This focus highlighting is useful for those navigating by keyboard, so shouldn't be hidden.
2. Navigating across a tablist using arrow keys doesn't currently work if there are next/previous buttons on the page (i.e. `jupyter-book`s), because of an event listener in `sphinx`'s `doctools.js` triggers the page to change. I've raised a separate PR to disable this when `button` elements are focused, like those used for tabs here.

**Comment by pradyunsg** [comment]
Ahoy! As a heads up, I won't be properly reviewing this myself anytime soon (context [here](https://github.com/executablebooks/meta/issues/145#issuecomment-708667465)) but I definitely wanted to drop in and say -- Thanks a lot for picking this up, and as far as I can see, this looks great!

**Comment by foster999** [comment]
Hey @chrisjsewell, please could you provide a review of this sometime? I appreciate that there's quite a few changes piled in here, but it made sense to include them given the extent of the rewrite.

If there's nothing obviously broken here I'll be around to pick up any bugs I may have thrown in, if released 😅

**Comment by Daltz333** [comment]
There seems to be some weird white outline of the tabbed content that just doesn't fit. 
image

**Comment by foster999** [comment]
@Daltz333 thanks for spotting that - I'd only tested with a white background... Only the open tab should have a background now

**Comment by chrisjsewell** [comment]
Heya soz for the delayed response. I will try to have a look at this in the next few days

**Comment by foster999** [comment]
No rush at all 👍

**Comment by Daltz333** [comment]
@chrisjsewell Any update on this? I'd love to use sphinx-tabs with sphinx-panels.

File: sphinx_tabs/static/doctools.js
Comment (chrisjsewell) [inline sphinx_tabs/static/doctools.js:1]: I'm a little tentative on overriding core aspects of sphinx 😬 (in case it is changed in some way upstream) are the modifications actually specific to sphinx-tabs or could you make a PR to sphinx to add these modifications?
At the least could you add a link to where the original document can be viewed (on https://github.com/sphinx-doc/sphinx) and clearly comment where the modifications are and why they have been made.

**Review by chrisjsewell — CHANGES_REQUESTED** [review]

**Comment by chrisjsewell** [comment]
Heya, @Daltz333 I don't want to be too much of a blocker on this package, so I've invited you and @foster999 as maintainers

I've made a quick review comment above also:

> It also includes an equivalent of #73

Could you add an example of this to the documentation (https://sphinx-tabs--94.org.readthedocs.build/en/94/#), and in fact it would be good to improve this page with a bit more explanation of the extension, including sphinx configuration options etc

File: sphinx_tabs/static/doctools.js
Comment (Daltz333) [reply sphinx_tabs/static/doctools.js:1]: Some pretty short research shows that it's near identical to https://github.com/sphinx-doc/sphinx/blob/3.x/sphinx/themes/basic/static/doctools.js

Comment (Daltz333) [reply sphinx_tabs/static/doctools.js:1]: Actually, it looks like they are identical, not sure why it's overriding?

Comment (foster999) [reply sphinx_tabs/static/doctools.js:1]: I'd been using this fork, so it was meant as a short term workaround really. But sphinx-doc/sphinx#8316 included the minor change in sphinx, so I'll drop it here 👍

Comment (foster999) [reply sphinx_tabs/static/doctools.js]: Removed this file from the PR

**Comment by foster999** [comment]
Thanks @chrisjsewell, sounds good to me

I've dropped the sphinx js override and updated the RTD docs to include code examples and describe features. Please could one of you have a quick look at https://sphinx-tabs--94.org.readthedocs.build/en/94/#, then I think it should be good to go 😄

**Review by Daltz333 — APPROVED** [review]
Looks fine to me! I think we should ship it if no other objections!

File: docs/index.rst
Comment (chrisjsewell) [inline docs/index.rst:24]: you are trying to use markdown in an RST file here 😉 

should be e.g.

```
.. code-block:: bash

   pip install sphinx-tabs

```

**Review by chrisjsewell — CHANGES_REQUESTED** [review]
thanks @foster999 looking good, just a minor thing to change

File: docs/index.rst
Comment (Daltz333) [reply docs/index.rst:24]: Ah, I didn't notice, as it renders partially correct on the actual site.
image

Comment (chrisjsewell) [reply docs/index.rst:24]: yeh I know, I just notice the extra `` ` `` and though something was a miss

Comment (foster999) [reply docs/index.rst:24]: 🤦🏻‍♂️ I copied that straight from the README without thinking...

**Comment by chrisjsewell** [comment]
cheers!

**Comment by chrisjsewell** [comment]
I imagine this should be released as v2.0.0

**Comment by pradyunsg** [comment]
Hurray! Thanks for doing this folks! ^>^

**Comment by foster999** [comment]
Cheers for the review both! Shall punch a new release now

**Comment by chrisjsewell** [comment]
I assume you know how: https://github.com/executablebooks/.github/blob/master/CONTRIBUTING.md#the-process-of-creating-a-release

**Pull request merged**
